planner/core/preprocess_test.go

```diff
@@ -197,6 +197,7 @@ func (s *testValidatorSuite) TestValidator(c *C) {
 		{"select * from (select 1 ) a , (select 2) b, (select * from (select 3) a join (select 4) b) c;", false, nil},
 
 		{"CREATE VIEW V (a,b,c) AS SELECT 1,1,3;", false, nil},
+		{"select cast(1 as binary(1024));", false, errors.New("[types:1074]Field length too big for target type (max = 255)")},
```

Comment:
Did you try this in mysql 5.7? Seems we get the expected result, not errors.
BTW, in mysql 8.0.
```sql

MySQL [(none)]> select cast(1 as binary(4294967295));
+-------------------------------+
| cast(1 as binary(4294967295)) |
+-------------------------------+
| NULL                          |
+-------------------------------+
1 row in set, 1 warning (0.00 sec)

MySQL [(none)]> show warnings;
+---------+------+--------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                              |
+---------+------+--------------------------------------------------------------------------------------+
| Warning | 1301 | Result of cast_as_binary() was larger than max_allowed_packet (67108864) - truncated |
+---------+------+--------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
I like the MySQL 8.0 behavior (check if the length is greater than `max_allowed_packet` and then return `NULL` + a warning).  I don't think the 5.7 behavior is intended.

Many [MySQL 8.0 features](https://github.com/pingcap/tidb/issues/7968) are already on our roadmap, so it's also not a stretch to think we may change our targeted compatible version.